src/dotnet-interactive/Connection/StdIoKernelConnector.cs

```diff
@@ -133,14 +134,15 @@ public async Task<ProxyKernel> CreateRootProxyKernelAsync()
             activity.Info("Process id: {0}", _process.Id);
 
             _receiver = KernelCommandAndEventReceiver.FromObservable(stdOutObservable);
-
-            bool kernelReadyReceived = false;
+            
+            _kernelReady = null;
             _receiver.Select(coe => coe.Event)
-                                   .OfType<KernelReady>()
+                                       .OfType<KernelReady>()
                                    .Take(1)
                                    .Subscribe(e =>
                                    {
-                                       kernelReadyReceived = true;
+                                       _kernelReady = e;
```

Comment:
Since this is all within the same method, could we use a local instead of the field for `kernelReady`?

We already cache all kernel infos that we have seen so far in `_remoteKernelInfos`. So, it feels a bit strange that at least some of those kernel infos would also be avbailable via the `_kernelReady` field.

Reply:
(Looks like making the above change would also solve the nullability error that caused the build break in CI)

Reply:
but the connector can be reused that is why is stored, you will see that event only when the process is started

Reply:
All kernel infos are already avaialble in _remoteInfos, so you could just fetch it from there (like we do inside CreateProxyKernelAsync below, could you not?